lib/contractWrapperFactory.ts

```diff
@@ -49,7 +75,67 @@ export class ContractWrapperFactory<TWrapper extends IContractWrapperBase>
    */
   public async deployed(): Promise<TWrapper> {
     await this.ensureSolidityContract();
-    return new this.wrapper(this.solidityContract, this.web3EventService).hydrateFromDeployed();
+
+    const getWrapper = (): Promise<TWrapper> => {
+      return new this.wrapper(this.solidityContract, this.web3EventService).hydrateFromDeployed();
+    };
+
+    return this.getHydratedWrapper(getWrapper);
+  }
+
+  private async getHydratedWrapper(
+    getWrapper: () => Promise<TWrapper>,
+    address?: Address): Promise<TWrapper> {
+
+    let hydratedWrapper: TWrapper;
+    if (ContractWrapperFactory.configService.get("cacheContractWrappers")) {
+      if (!address) {
+        try {
+          address = this.solidityContract.address;
+        } catch {
+          // the contract has not been deployed, so can't get it's address
+        }
+      }
+
+      if (address) {
+        hydratedWrapper = this.getCachedContract(this.solidityContractName, address) as TWrapper;
```

Comment:
should maybe log if the returned contract is coming from the cache or not?

Reply:
yeah, I was wanting to do that too.  done.